### ISSUE TITLE
Remove with backspace and delete not working when format attribute is enabled

### ISSUE THREAD

**Issue by DuudeXX8** [s-yadav/react-number-format#452]
Hello, 

I've noticed a bug with the `Delete ` and `Backspace ` key handling. If format attribute is enabled user can only  remove all characters from input.He can't highlight and remove part of numbers. `Delete `and `Backspace `does not work.

```
<NumberFormat
          value={this.state.decimals}
          onValueChange={this.handleChange("decimals")}
          decimalScale={0}
          customInput={TextField}
          variant="outlined"
          format="## ### ## ##" // if we remove this attribute it will work fine
          label="Decimals"
/>

```
[**Codesandbox Demo**](https://codesandbox.io/s/numberformat-format-forked-nehd6?file=/src/demo.js)

![captured (2)](https://user-images.githubusercontent.com/19229161/97966168-e52aed00-1dd4-11eb-929b-79f6d5fc81a1.gif)

**Any ideas?**

**Comment by jologz** [comment]
We are experiencing this issue.  I created a PR to resolve. hopefully, it gets reviewed.

**Comment by jasan-s** [comment]
same here

**Comment by huythangphan** [comment]
@s-yadav can you solve this?

**Comment by AlexR0v** [comment]
there was the same problem. Can you tell me how to solve it?

**Comment by dayoolacodes** [comment]
has this been resolved?

**Comment by jologz** [comment]
No, it has not.  The author won't approve my PR so I just created our own component, instead.

**Comment by s-yadav** [comment]
I will take a look into the PR today. Probably I need to set a reminder on this things. 😔

**Comment by dayoolacodes** [comment]
ok great

**Comment by s-yadav** [comment]
Hey @jologz, tested your code, and everything looks working. I have left some minor comments. Can you check that? And then the PR is good to go.

**Comment by dayoolacodes** [comment]
Hi @s-yadav, I also notice that a long keypress on backspace does not clear the input field. Will this PR solve that issue also?

**Comment by dayoolacodes** [comment]
Hi @s-yadav, I also noticed that a long keypress on the backspace key does not clear the input field. Will this PR solve that issue also?

**Comment by s-yadav** [comment]
Hummm, It could be. let's check once I make a new release.

**Comment by s-yadav** [comment]
This is available on 4.6.1 now.
